clang/lib/Sema/SemaChecking.cpp

```diff
@@ -8469,7 +8489,7 @@ void Sema::CheckInfNaNFunction(const CallExpr *Call,
     bool IsInfOrIsFinite =
```

Comment:
As an aside, this is called on every function call, which seems widely inficient.

can we do something like
```cpp
FPOptions FPO = Call->getFPFeaturesInEffect(getLangOpts());
if(FPO.getNoHonorInfs())
  return;
```

(and remove `FPO.getNoHonorInfs()` in the rest of the function)

Reply:
Yes, I agree, i am thinking to change the structure of the function `CheckInfNaNFunction ` too, because if `HasIdentifier` is false that's means `IsStdFunction`, `IsInfOrNanFunction` and `IsInfinityFunction` will return false, so maybe we can just eliminate some un needed function calls by early return.

I will try and update you with result :D

Reply:
I think it's better now, early return if there is no identifier and check `NoHonor.. `first before checking for function name